buildSrc/src/test/java/io/bitrise/trace/internal/UpdateChangeLogTaskTest.java

```diff
@@ -0,0 +1,402 @@
+    private static final String dummyCommitFooter3 = "APM-34567";
+    private static final String dummyCommitMessage3 = String.format(dummyCommitTemplate, dummyCommitType3,
+            dummyCommitTitle3, dummyCommitDetails3, dummyCommitFooter3);
+
+    private static final String dummyCommitType4 = "feat!";
+    private static final String dummyCommitTitle4 = "Breaking change";
+    private static final String dummyCommitDetails4 = "API break, new API enables everything!\nUse it " +
+            "wisely!\nDeprecated some things";
+    private static final String dummyCommitFooter4 = "APM-45678";
+    private static final String dummyCommitMessage4 = String.format(dummyCommitTemplate, dummyCommitType4,
+            dummyCommitTitle4, dummyCommitDetails4, dummyCommitFooter4);
+
+    private static final String dummyCommitType5 = "chore";
+    private static final String dummyCommitTitle5 = "CI update";
+    private static final String dummyCommitDetails5 = "New bitrise.yml";
+    private static final String dummyCommitFooter5 = "APM-56789";
+    private static final String dummyCommitMessage5 = String.format(dummyCommitTemplate, dummyCommitType5,
+            dummyCommitTitle5, dummyCommitDetails5, dummyCommitFooter5);
+
+    private static final String dummyTagName1 = "0.0.1";
+    private static final String dummyTagName2 = "0.1.0";
+
+    private final String dummyReleaseName = "Dummy release";
+    private final List<String> changeLogLines = new ArrayList<String>() {{
+        add("CHANGES");
+        add("=======");
+        add("");
+    }};
+
+    @BeforeClass
+    public static void setupClass() throws GitAPIException, IOException {
+        setUpDummyRepo();
+    }
+    //endRegion
+
+    // region CommitTypes
+
+    @Test
+    public void getMajorCommitTypes_ShouldReturnExpectedValues() {
+        final Set<String> actual = UpdateChangeLogTask.getMajorCommitTypes();
+        final Set<String> expected = new HashSet<>(Arrays.asList("fix!", "feat!"));
+        assertEquals(actual, expected);
+    }
+
+    @Test
+    public void getMinorCommitTypes_ShouldReturnExpectedValues() {
+        final Set<String> actual = UpdateChangeLogTask.getMinorCommitTypes();
+        final Set<String> expected = new HashSet<>(Collections.singletonList("feat"));
+        assertEquals(actual, expected);
+    }
+
+    @Test
+    public void getPatchCommitTypes_ShouldReturnExpectedValues() {
+        final Set<String> actual = UpdateChangeLogTask.getPatchCommitTypes();
+        final Set<String> expected = new HashSet<>(Collections.singletonList("fix"));
+        assertEquals(actual, expected);
+    }
+
+    @Test
+    public void getAllowedCommitTypes_ShouldReturnExpectedValues() {
+        final Set<String> actual = UpdateChangeLogTask.getAllowedCommitTypes();
+        final Set<String> expected = new HashSet<>(Arrays.asList("fix", "feat", "fix!", "feat!"));
+        assertEquals(actual, expected);
+    }
+    // endRegion
+
+    // region getting commits
```

Comment:
These tests in this region create a repo in a temporary folder with some dummy commits, this way it will be reproducible.